caddyhttp/httpserver/logger.go

```diff
@@ -64,6 +67,25 @@ func (l Logger) Printf(format string, args ...interface{}) {
 	l.fileMu.RUnlock()
 }
 
+func (l Logger) MaskIP(ip string) string {
+	var myIP, maskIP net.IP
+
+	// If unable to parse, simply return IP as provided.
+	myIP = net.ParseIP(ip)
+	if myIP == nil {
+		return ip
+	}
+
+	if myIP.To4() == nil {
+		maskIP = net.ParseIP(l.V6ipMask)
```

Comment:
Is it feasible to move the `net.ParseIP` calls into the setup function? This would waste CPU cycles on each request.

Side note: if `l.V4ipMask`/`l.V6ipMask` were of type `*net.IP`, this would allow for an early return avoiding the `net.IPMask` call. Useful, when no masking should happen.